src/include/migraphx/generate.hpp

```diff
@@ -36,36 +36,38 @@ inline namespace MIGRAPHX_INLINE_NS {
 template <class T, MIGRAPHX_REQUIRES(is_floating_point<T>{})>
 constexpr T normalize(unsigned long z)
 {
-    if(z == 0)
-        return T(0);
-    const auto max     = 32;
-    const double range = max / 2; // NOLINT
-    double result      = double(z % max) / range;
-    result -= 1;
+    const auto max     = 1ULL << (sizeof(T) * 8 - 1);
+    const double range = max / 2.0;
+    double result      = -1.0 + (z % max) / range;
     return T(result);
+    // Expected output: between -1.0 and 1.0
 }
 
 template <class T, MIGRAPHX_REQUIRES(is_signed<T>{} and not is_floating_point<T>{})>
 constexpr T normalize(unsigned long z)
 {
-    const auto max      = 1ULL << (sizeof(T) * 5);
+    const long long max = 1ULL << (sizeof(T) * 8 - 2);
```

Comment:
This should not be changed. This will lead to overflow in the tests which can work differently for different platforms making verification impossible. The `5` factor was used to keep some bits free so we can multiply or add these numbers together.

Reply:
Add an inline comment explaining why the value exists.  Feel free to copy paste

Reply:
This has been analyzed, and a suitable expanded value has been added to code. All the debug/non-debug builds are passing.